ecommerce_extensions/tenant/extra_options.py

```diff
@@ -0,0 +1,60 @@
+"""
+This function gets called during every request by the
+extra_html and extra_scripts to validate all the custom html and scripts for a specific path.
+"""
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+def validate_option_attributes(values, common_attributes, error_message):
+    """
+    Validate common attributes for options extra.
+
+    Parameters:
+        values (dict): a option with values to validate according to common_attributes
+
+    Returns:
+        dict: a option with values validated according to common_attributes
+    """
+    option_attributes = {}
+
+    for key, attr_values in common_attributes.items():
+
+        value = values.get(key)
+
+        try:
+            if value is not None:
+                attr_values['options'].index(value.lower())
+                option_attributes[key] = value.lower()
+            else:
+                default = attr_values['default']
+                option_attributes[key] = attr_values['options'][default]
+        except (KeyError, ValueError):
+            logger.error(error_message, key)
+
+            return None
+
+    return option_attributes
+
+
+def check_attribute(values, option_attributes, attribute, error_message):
```

Comment:
why pass the error message as a param?
I think **"HTML could not get loaded. '%s' attribute is missing or is an invalid option."** it's a pretty standard message and we could leave that hardcoded inside check_attribute instead of passing it as a param.